controllers/remote/cluster_cache.go

```diff
@@ -154,8 +159,15 @@ func (t *ClusterCacheTracker) newClusterAccessor(ctx context.Context, cluster cl
 		cancelFunc: cacheCtxCancel,
 	}
 
+	if err := cache.IndexField(ctx, &corev1.Node{}, NodeProviderIDIndex, indexNodeByProviderID); err != nil {
+		return nil, fmt.Errorf("failed to index field %s: %w", NodeProviderIDIndex, err)
+	}
```

Comment:
Should we create this indexer for every client?

Reply:
It's being created for each newClusterAccessor, which others are you thinking of?

Reply:
Given that this is a remote client, we're making an assumption here that we'd get nodes and we want to index them

Reply:
It seems reasonable to me, if you disagree we could move the indexing logic to the Watch method.

Reply:
We should move this under the `Watch` method maybe? We should only ever create this index if a `&corev1.Node{}` is passed in the WatchInput struct as `Kind` object.